packages/dtslint/src/rules/preferDeclareFunctionRule.ts

```diff
@@ -1,36 +0,0 @@
-import * as Lint from "tslint";
-import * as ts from "typescript";
-
-import { eachModuleStatement, failure } from "../util";
-
-export class Rule extends Lint.Rules.AbstractRule {
-  static metadata: Lint.IRuleMetadata = {
-    ruleName: "prefer-declare-function",
-    description: "Forbids `export const x = () => void`.",
-    optionsDescription: "Not configurable.",
-    options: null,
-    type: "style",
-    typescriptOnly: true,
-  };
-
-  static FAILURE_STRING = failure(
-    Rule.metadata.ruleName,
-    "Use a function declaration instead of a variable of function type."
-  );
-
-  apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-    return this.applyWithFunction(sourceFile, walk);
-  }
-}
-
-function walk(ctx: Lint.WalkContext<void>): void {
-  eachModuleStatement(ctx.sourceFile, (statement) => {
```

Comment:
I missed this call to eachModuleStatement. It

1. Only runs on statements in declaration files.
2. Recursively runs on `declare module "foo"` blocks. This includes namespaces too, as far as I can tell.

The selector on line 22 is probably good enough for (2), but I'm not sure how to make (1) happen yet.